js/bitfinex.js

```diff
@@ -541,25 +542,35 @@ module.exports = class bitfinex extends Exchange {
                 baseId = id.slice (0, 3);
                 quoteId = id.slice (3, 6);
             }
-            const base = this.safeCurrencyCode (baseId);
-            const quote = this.safeCurrencyCode (quoteId);
+            const splitBase = baseId.split ('f0');
+            const splitQuote = quoteId.split ('f0');
+            const baseRaw = this.safeString (splitBase, 0);
+            const quoteRaw = this.safeString (splitQuote, 0);
+            const base = this.safeCurrencyCode (baseRaw);
+            const quote = this.safeCurrencyCode (quoteRaw);
+            let symbol = base + '/' + quote;
+            let settle = undefined;
+            if (swap) {
+                settle = quote;
+                symbol = symbol + ':' + settle;
+            }
             result.push ({
                 'id': id,
-                'symbol': base + '/' + quote,
+                'symbol': symbol,
                 'base': base,
                 'quote': quote,
                 'settle': undefined,
                 'baseId': baseId,
                 'quoteId': quoteId,
                 'settleId': undefined,
-                'type': 'spot',
-                'spot': true,
+                'type': type,
+                'spot': spot,
                 'margin': this.safeValue (market, 'margin'),
-                'swap': false,
-                'future': false,
+                'swap': swap,
+                'future': future,
                 'option': false,
                 'active': true,
-                'contract': false,
+                'contract': future,
                 'linear': undefined,
```

Comment:
We will need to indicate whether a contract is linear or inverse here, otherwise that contract isn't entirely usable, it depends on the settle currency which we must also fill-in somehow.